tests/unit/plugins/lookup/onepassword/test_onepassword.py

```diff
@@ -0,0 +1,179 @@
+        op.set_token()
+
+    op._cli.full_signin.assert_not_called()
+
+
+@pytest.mark.parametrize("op_fixture", OP_VERSION_FIXTURES)
+def test_op_set_token_with_config_full_signin(op_fixture, request, mocker):
+    op = request.getfixturevalue(op_fixture)
+    mocker.patch("os.path.isfile", return_value=True)
+    mocker.patch.object(op._cli, "signin", return_value=(99, "", ""), side_effect=AnsibleLookupError("Raised intentionally"))
+    mocker.patch.object(op._cli, "full_signin", return_value=(0, "", ""))
+
+    op.set_token()
+
+    op._cli.full_signin.assert_called()
+
+
+@pytest.mark.parametrize("op_fixture", OP_VERSION_FIXTURES)
+def test_op_set_token_without_config(op_fixture, request, mocker):
+    op = request.getfixturevalue(op_fixture)
+    token = "B988E8A2680A4A348962751A96861FA1"
+    mocker.patch("os.path.isfile", return_value=False)
+    mocker.patch.object(op._cli, "signin", return_value=(99, "", ""))
+    mocker.patch.object(op._cli, "full_signin", return_value=(0, token + "\n", ""))
+
+    op.set_token()
+
+    op._cli.signin.assert_not_called()
+    assert op.token == token
+
+
+@pytest.mark.parametrize(
+    ("op_fixture", "login_status"),
+    [(op, value) for op in OP_VERSION_FIXTURES for value in [False, True]]
+)
+def test_op_assert_logged_in(mocker, login_status, op_fixture, request):
+    op = request.getfixturevalue(op_fixture)
+    mocker.patch.object(op._cli, "assert_logged_in", return_value=login_status)
+    mocker.patch.object(op, "set_token")
+
+    op.assert_logged_in()
+
+    op._cli.assert_logged_in.assert_called_once()
+    assert op.logged_in == login_status
+
+    if not login_status:
+        op.set_token.assert_called_once()
+
+
+@pytest.mark.parametrize("op_fixture", OP_VERSION_FIXTURES)
+def test_op_get_raw_v1(mocker, op_fixture, request):
+    op = request.getfixturevalue(op_fixture)
+    mocker.patch.object(op._cli, "get_raw", return_value=[99, "RAW OUTPUT", ""])
+
+    result = op.get_raw("some item")
+
+    assert result == "RAW OUTPUT"
+    op._cli.get_raw.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("op_fixture", "output", "expected"),
+    (
+        list(itertools.chain([op], d))
+        for op in OP_VERSION_FIXTURES
+        for d in [
+            ("RAW OUTPUT", "RAW OUTPUT"),
+            (None, ""),
+            ("", ""),
+        ]
+    )
+)
+def test_op_get_field(mocker, op_fixture, output, expected, request):
+    op = request.getfixturevalue(op_fixture)
+    mocker.patch.object(op, "get_raw", return_value=output)
+    mocker.patch.object(op._cli, "_parse_field", return_value=output)
+
+    result = op.get_field("some item", "some field")
+
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    ("cli_class", "vault", "queries", "kwargs", "output", "expected"),
+    (
+        (_cli_class, item["vault_name"], item["queries"], item.get("kwargs", {}), item["output"], item["expected"])
+        for _cli_class in MOCK_ENTRIES
```

Comment:
This might help againts the 'strange' unit test errors:
```suggestion
        for _cli_class in sorted(MOCK_ENTRIES)
```

Reply:
I've been fighting with that for several weeks and didn't have any luck with `sorted()`. I just pushed a change generating IDs that may help. If not, I'll give your suggestion a try.

Reply:
It looks very much like a 'dictionary order is variable' problem with Python < 3.6, so *some* `sorted` should help. I was hoping this might be the right place :)

Reply:
Looks like my solution to name things worked but I like your simple fix better.

Reply:
I haven't tested mine, so maybe it doesn't even work. Anyway, I'm glad it's now fixed :)

Reply:
It seems like this is still happening: https://dev.azure.com/ansible/community.general/_build/results?buildId=59661&view=logs&j=f8bcbeef-cfea-5c1d-f0dd-e093f18ffc7a&t=92f9f916-b1da-5037-50d9-d4b0bb191425&s=020dae3b-9992-5006-b9e9-1114a7f5135d
Maybe reverting f5e10a8f422952ff831c21101596082ce0571d9a might be the better solution?